### ISSUE TITLE
[Question] How should i handle alerts when i use Page Object Model?

### ISSUE THREAD

**Issue by SRVFLLN** [microsoft/playwright#18249]
Hello! When i click on button that triggers alert, nothing happens, and i can't figure out why. I have tryed differrent ways to set page.on('dialog'), but nothing works.
My spec file:
![image](https://user-images.githubusercontent.com/93190152/197232308-bc20d3d8-af4f-4435-af00-b6fb0d9b4b1c.png)

Page obect file: 
![image](https://user-images.githubusercontent.com/93190152/197232560-2476d121-090c-402a-b747-6245e86c0e6c.png)

Can you help me understand where my problem is?

**Comment by devgony** [comment]
Did you get what was the problem?

**Comment by SRVFLLN** [comment]
> Did you get what was the problem?

Yes, if i remember correctly, I forgot to add "await" in line, which triggered alert. So, i checked all asynchronous code, and then it worked.
